types/encoding-japanese/index.d.ts

```diff
@@ -26,7 +27,7 @@ type IntArrayType =
     | Int8Array
     | Int16Array
     | Int32Array;
-type RawType = IntArrayType | ReadonlyArray<number> | Buffer;
+type RawType = IntArrayType | ReadonlyArray<number> | Uint8Array;
```

Comment:
This package supports `Buffer`. So it should not be removed. Since this package supports Node.js explicitly, `node.d.ts` dependency is necessary.

https://www.npmjs.com/package/encoding-japanese

> Each methods are also available for the Buffer in Node.js.

Reply:
And `Uint8Array` is already included in `IntArrayType`. So it should not be added to variants of `RawType`.

Reply:
> And `Uint8Array` is already included in `IntArrayType`. So it should not be added to variants of `RawType`.

ops, sorry, amended

Reply:
> This package supports `Buffer`. So it should not be removed. Since this package supports Node.js explicitly, `node.d.ts` dependency is necessary.

Yes, I understand, it still works perfectly fine under CJS/Node. the types do not rely on specific Buffer methods are are not in the Uint8Array, so I would like to remove the burden put on non-CJS users to install @types/node, just to compile code that works via esm or global script. So on usage it works perfectly fine (compiles and works at runtime) with a Buffer:

```ts
import Encoding = require("encoding-japanese");

const buffer = Buffer.from([1, 2, 3]);
console.log(Encoding.convert(buffer, "SJIS", "UTF8"));
console.log(Encoding.convert(buffer, "UTF16", "UTF8"));
console.log(Encoding.convert(buffer, "UTF16LE", "UTF8"));
console.log(Encoding.convert(buffer, "UTF16BE", "UTF8"));
console.log(Encoding.convert(buffer, "BINARY", "UTF8"));
console.log(Encoding.convert(buffer, "UTF32", "UTF8"));
console.log(Encoding.convert(buffer, "AUTO", "UTF8"));
console.log(Encoding.detect(buffer));
console.log(
  Encoding.convert(buffer, {
    to: "UTF16",
    from: "UTF8",
    bom: true,
  })
);
```

Reply:
Thanks for the explanation. So it relies on that `Buffer` is a superset of `Uint8Array`, am I correct?

Would you please add test case which uses node's `Buffer` to confirm it works?

Reply:
I'll do this, just reminder this will increase this time spent on DT tests again.

Reply:
Thanks. I'll wait for the update.

Reply:
@rhysd amended, thx! Sorry for delay, btw.

Reply:
Thank you for the update. I'll review again.